nukebuild/BuildParameters.cs

```diff
@@ -0,0 +1,186 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Xml.Linq;
+using Nuke.Common;
+using Nuke.Common.BuildServers;
+using Nuke.Common.Execution;
+using Nuke.Common.IO;
+using static Nuke.Common.IO.FileSystemTasks;
+using static Nuke.Common.IO.PathConstruction;
+using static Nuke.Common.Tools.MSBuild.MSBuildTasks;
+
+public partial class Build
+{
+    [Parameter("configuration")]
+    public string NukeArgConfiguration { get; set; }
+    
+    [Parameter("skip-tests")]
+    public bool NukeArgSkipTests { get; set; }
+    
+    [Parameter("force-nuget-version")]
+    public string NukeArgForceNugetVersion { get; set; }
+    
+    public class BuildParameters
+    {
+        public string Configuration { get; }
+        public bool SkipTests { get; }
+        public string MainRepo { get; }
+        public string MasterBranch { get; }
+        public string RepositoryName { get; }
+        public string RepositoryBranch { get; }
+        public string ReleaseConfiguration { get; }
+        public string ReleaseBranchPrefix { get; }
+        public string MSBuildSolution { get; }
+        public bool IsLocalBuild { get; }
+        public bool IsRunningOnUnix { get; }
+        public bool IsRunningOnWindows { get; }
+        public bool IsRunningOnAppVeyor { get; }
+        public bool IsRunningOnAzure { get; }
+        public bool IsPullRequest { get; }
+        public bool IsMainRepo { get; }
+        public bool IsMasterBranch { get; }
+        public bool IsReleaseBranch { get; }
+        public bool IsTagged { get; }
+        public bool IsReleasable { get; }
+        public bool IsMyGetRelease { get; }
+        public bool IsNuGetRelease { get; }
+        public bool PublishTestResults { get; }
+        public string Version { get; }
+        public AbsolutePath ArtifactsDir { get; }
+        public AbsolutePath NugetRoot { get; }
+        public AbsolutePath ZipRoot { get; }
+        public AbsolutePath BinRoot { get; }
+        public AbsolutePath TestResultsRoot { get; }
+        public string DirSuffix { get; }
+        public List<string> BuildDirs { get; }
+        public string FileZipSuffix { get; }
+        public AbsolutePath ZipCoreArtifacts { get; }
+        public AbsolutePath ZipNuGetArtifacts { get; }
+        public AbsolutePath ZipSourceControlCatalogDesktopDir { get; }
+        public AbsolutePath ZipTargetControlCatalogDesktopDir { get; }
+
+
+       public BuildParameters(Build b)
+        {
+            var buildSystem = Host;
```

Comment:
nit: rename uses of `buildSystem` to use `Host` and remove the `buildSystem` variable.